lnwallet/channel.go

```diff
@@ -3626,6 +3672,11 @@ func (lc *LightningChannel) validateCommitmentSanity(theirLogCounter,
 	ourInitialBalance := commitChain.tip().ourBalance
 	theirInitialBalance := commitChain.tip().theirBalance
 
+	// Initiate feePerKw to the last committed fee for this chain as we'll
+	// need this to determine which HTLCs are dust, and also the final fee
+	// rate.
+	view.feePerKw = commitChain.tip().feePerKw
```

Comment:
This seems incorrect. We only know the final fee rate for this view after we fully evaluate it (as we do below).

Reply:
Yeah, reverted this, as it was the culprit of the test failures